### PR TITLE
Add dynamic tasks support

### PR DESCRIPTION
**Pull request by ognjenkatic**

### CODE REVIEW

File: test/ConductorSharp.Engine.Tests/Samples/Workflows/DecisionInDecision.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
                   "startDelay": 0,
                   "subWorkflowParam": {
                     "name": "NOTIFICATION_send_to_customer",
-                    "version": 1,
+                    "version": "1",
```
Comment (boma96) [inline test/ConductorSharp.Engine.Tests/Samples/Workflows/DecisionInDecision.json:69]: Why is this changed to string?

File: src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs [hunk 2]
```diff
@@ -156,6 +156,11 @@ string script
             Expression<Func<TWorkflow, F>> input
         ) where F : IRequest<G> => AddAndReturnBuilder(new JsonJqTransformTaskBuilder<F, G>(refference.Body, input.Body));
 
+        public ITaskOptionsBuilder AddTask<F, G>(
```
Comment (boma96) [inline src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs:159]: Let's try giving these type parameters meaningful names for new code. I suggest `TInput `and `TOutput`

File: src/ConductorSharp.Client/Model/Common/WorkflowDefinition.cs [hunk 3]
```diff
@@ -14,8 +14,21 @@ public class SubWorkflowParam
...
+                        return version;
+
+                    return 0;
+                }
+                set { VersionAsString = value.ToString(); }
+            }
+
             [JsonProperty("version")]
-            public int Version { get; set; }
+            public string VersionAsString { get; set; }
```
Comment (boma96) [inline src/ConductorSharp.Client/Model/Common/WorkflowDefinition.cs:31]: Why is this changed to string?

File: src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs [hunk 4]
```diff
@@ -156,6 +156,11 @@ string script
...
+            Expression<Func<TWorkflow, DynamicTaskModel<F, G>>> refference,
+            Expression<Func<TWorkflow, DynamicTaskInput<F, G>>> input
+        ) => AddAndReturnBuilder(new DynamicTaskBuilder<F, G>(refference.Body, input.Body));
```
Comment (boma96) [inline src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs:162]: `reference`

File: src/ConductorSharp.Engine/Builders/DynamicTaskBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Text;
+
+namespace ConductorSharp.Engine.Builders
+{
+    public class DynamicTaskBuilder<I, O> : BaseTaskBuilder<DynamicTaskInput<I, O>, O>
+    {
+        public const string TaskType = "DYNAMIC";
```
Comment (boma96) [inline src/ConductorSharp.Engine/Builders/DynamicTaskBuilder.cs:14]: These can be private if they are only used in this class

File: src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs (hunk 2 above)
Comment (ognjenkatic) [reply src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs:159]: Agreed, but i suggest doing this revamp in a different PR where we tidy up these things

File: src/ConductorSharp.Client/Model/Common/WorkflowDefinition.cs (hunk 3 above)
Comment (ognjenkatic) [reply src/ConductorSharp.Client/Model/Common/WorkflowDefinition.cs:31]: Dynamic tasks can have their version passed in as a parameter bound from a task or workflow input so we must support both options.

File: src/ConductorSharp.Engine/Builders/DynamicTaskBuilder.cs (hunk 5 above)
Comment (ognjenkatic) [reply src/ConductorSharp.Engine/Builders/DynamicTaskBuilder.cs:14]: For the moment they are only used in this class, so I will make them private. In the future we might have a reason to access this outside of the builder.

**Comment by ognjenkatic** [comment]
Merging as comments have been resolved
